rules/go_rules.build_defs

```diff
@@ -123,6 +123,9 @@ def go_library(name:str, srcs:list, resources:list=None, asm_srcs:list=None, hdr
       package (str): The package as it would appear at the top of the go source files. Defaults
                      to name.
     """
+    if CONFIG.FF_FAIL_ON_BAD_GLOB and srcs == []:
```

Comment:
This shouldn't happen here - it should be a general check in the `glob` implementation in `builtins.go`

Reply:
I was unsure about putting it in the `glob` implementation because I didn't know if we wanted to fail on _any_ bad glob. Sticking it in the go rules means it wouldn't be a breaking change I think as we already fail in that case, just with an unhelpful error message. But I'm happy to stick it in there if you think that's best - Jon is also of the opinion that no one would intentionally write a glob that doesn't match anything. Also agree on the opt-out - would soften the impact of the breaking.